File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/db/dao/TeamDao.kt
package dev.tpavani.octopoints.db.dao

import androidx.room.Dao
import androidx.room.Query
import dev.tpavani.octopoints.db.entity.TeamEntity

@Dao
interface TeamDao : IDao<TeamEntity>{
    @Query("SELECT * FROM teams WHERE matchId=:id")
    suspend fun getTeamsByMatchId(id: Int): List<TeamEntity>
}

File: octopoints_flutter/lib/db/dao/TeamDao.dart
import 'package:floor_annotation/floor_annotation.dart';
import 'package:octopoints_flutter/db/dao/IDao.dart';
import 'package:octopoints_flutter/db/entity/TeamEntity.dart';

@dao
abstract class TeamDao extends IDao<TeamEntity> {
  @Query('SELECT * FROM teams WHERE matchId=:id ORDER BY total DESC')
  Future<List<TeamEntity>> getTeamsByMatchId(int id);
}
